remote-api/src/test/java/org/alfresco/rest/api/tests/NodeSizeDetailsTest.java

```diff
@@ -0,0 +1,254 @@
+        return ciExpected;
+    }
+
+    private String addToDocumentLibrary(Site testSite, String name, String nodeType)
+    {
+        String parentId;
+        try
+        {
+            parentId = getSiteContainerNodeId(testSite.getId(), "documentLibrary");
+            return createNode(parentId, name, nodeType, null).getId();
+        }
+        catch (Exception e)
+        {
+            LOG.error("Exception occurred in NodeSizeDetailsTest:addToDocumentLibrary {}", e.getMessage());
+        }
+        return null;
+    }
+
+    @Before
+    public void setup() throws Exception
+    {
+        super.setup();
+        setRequestContext(user1);
+
+        String siteTitle = "RandomSite" + System.currentTimeMillis();
+        this.userOneN1Site = createSite("RN" + RUNID, siteTitle, siteTitle, SiteVisibility.PRIVATE, 201);
+
+        // Create a folder within the site document's library.
+        String folderName = "folder" + System.currentTimeMillis();
+        folderId = addToDocumentLibrary(userOneN1Site, folderName, TYPE_CM_FOLDER);
+        permissionService = applicationContext.getBean("permissionService", PermissionService.class);
+        nodes = applicationContext.getBean("Nodes", Nodes.class);
+    }
+
+    /**
+     * Test case for POST/size-details, which request and calculates the size of a folder. GET/size-details, which retrieve the SizeDetails of the folder Node. {@literal <host>:<port>/alfresco/api/<networkId>/public/alfresco/versions/1/nodes/<nodeId>/size-details} {@literal <host>:<port>/alfresco/api/<networkId>/public/alfresco/versions/1/nodes/<nodeId>/size-details/<jobId>}
+     */
+    @Test
+    public void testPostAndGetFolderSizeDetails() throws Exception
+    {
+        // Perform POST request
+        HttpResponse postResponse = post(generateNodeSizeDetailsUrl(folderId), null, 202);
+
+        assertNotNull("After executing POST/size-details first time, it will provide jobId with 202 status code",
+                postResponse.getJsonResponse());
+
+        JSONObject jsonObject = (JSONObject) postResponse.getJsonResponse()
+                .get("entry");
+
+        String jobId = (String) jsonObject.get("jobId");
+        assertNotNull("In response, JobId should be present", jobId);
+
+        // Prepare parameters.
+        Map<String, String> params = new HashMap<>();
+        params.put("nodeId", folderId);
+        params.put("jobId", jobId);
+
+        HttpResponse getResponse = getSingle(getNodeSizeDetailsUrl(folderId, jobId), null, 200);
+
+        assertNotNull("After executing GET/size-details, it will provide NodeSizeDetails with 200 status code",
+                getResponse.getJsonResponse());
+
+        NodeSizeDetails nodeSizeDetails = NodeSizeDetails.parseNodeSizeDetails(
+                (JSONObject) getResponse.getJsonResponse()
+                        .get("entry"));
+
+        assertNotNull("We are not getting correct response " + nodeSizeDetails, nodeSizeDetails.getStatus());
```

Comment:
Cen we add an assertion that the job finished and the calculated size is 0?

Reply:
We have created **createSizeDetails** in which we have used API to determine folder size details.